ui/src/pages/Edit/UploadStory.tsx

```diff
@@ -0,0 +1,721 @@
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import { DropzoneArea } from "material-ui-dropzone";
+import * as React from "react";
+import { useReducer } from "react";
+import { Redirect } from "react-router-dom";
+import styled from "styled-components";
+import useSWR from "swr";
+
+import { StoryDrawer } from "../../components";
+import { citiesList } from "../../data/cities";
+import { colors } from "../../styles/colors";
+import { device } from "../../styles/device";
+import {
+  UploadLabelsText,
+  UploadStoriesHeading,
+} from "../../styles/typography";
+import { Author, Story } from "../../types";
+import { ListboxComponent, renderGroup } from "./Listbox";
+import { INIT_STATE, uploadStoryReducer } from "./reducer";
+import { StoryProps, TagParameters } from "./types";
+
+const StyledGrid = styled(Grid)`
+  background-color: ${colors.primaryLight6};
+  @media ${device.laptop} {
+    justify: left;
+    width: 100vw;
+  }
+`;
+
+const StyledSelect = styled(Select)`
+  width: 30vw;
+  && {
+    font-family: Poppins;
+    align-items: center;
+    display: inline-block;
+  }
+`;
+
+const StyledMenuItem = styled(MenuItem)`
+  && {
+    font-family: Poppins;
+  }
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 30vw;
+
+  && {
+    margin-top: 12px;
+    .MuiInputLabel-formControl {
+      font-family: Poppins !important;
+    }
+  }
+`;
+
+const ImageContainer = styled.div`
+  width: 30vw;
+`;
+
+const StyledDropzoneArea = styled(DropzoneArea)`
+  && {
+    margin-top: 12px;
+  }
+`;
+
+const StyledAutocomplete = styled(Autocomplete)`
+  && {
+    font-family: Poppins;
+    margin-top: 12px;
+    width: 30vw;
+    .MuiInputLabel-root {
+      font-family: Poppins !important;
+    }
+    .MuiInputBase-input {
+      font-family: Poppins !important;
+    }
+  }
+`;
+
+const StyledButton = styled(Button)`
+  text-transform: none;
+  margin: 5px;
+`;
+
+const AddCountryPaper = styled(Paper)`
+  width: 30vw;
+  font-family: Poppins !important;
+`;
+
+const AddCountryDiv = styled.div`
+  margin-left: 18px;
+  margin-top: 5px;
+`;
+
+const AddCountryButton = styled(Button)`
+  margin-right: 5px;
+  width: 2vw;
+  float: right;
+  height: 3.75vh;
+
+  .MuiButton-label {
+    color: ${colors.primaryDark1};
+    font-family: Poppins !important;
+  }
+`;
+const StyledBackgroundColor = styled.div`
+  background-color: ${colors.white};
+  width: 40vw;
+  padding: 0px 0px 24px 24px;
+  margin-bottom: 24px;
+`;
+
+const StyledLinearProgress = styled(LinearProgress)`
+  margin-top: 24px;
+`;
+
+const StyledImage = styled.img`
+  width: 30vw;
+`;
+
+export const UploadStory: React.FC<StoryProps> = ({
+  id,
+  currentStory,
+  bio,
+}: StoryProps) => {
+  const { data: tagOptions, error } = useSWR<string[]>("/api/tags");
+  const { data: countries, error: errorCountries } = useSWR<string[]>(
+    "/api/countries"
+  );
+  const [state, dispatch] = useReducer(
+    uploadStoryReducer,
+    INIT_STATE(currentStory)
```

Comment:
If this is a function, lets not make it all uppercase. 
```suggestion
    get_init_state(currentStory)
```